### PR TITLE
Reference new full framework reference assemblies on NuGet

### PR DESCRIPTION
**Pull request by JosephWoodward**

### CODE REVIEW

File: src/Shouldly/Shouldly.csproj [hunk 1]
```diff
@@ -21,7 +20,7 @@
     <Optimize>false</Optimize>
   </PropertyGroup>
   <PropertyGroup Condition="'$(TargetFramework)'== 'net451'">
-      <FrameworkPathOverride>$(NuGetPackageRoot)microsoft.targetingpack.netframework.v4.5.1/1.0.1/lib/net451/</FrameworkPathOverride>
+      <FrameworkPathOverride>$(NuGetPackageRoot)microsoft.netframework.referenceassemblies.net451/1.0.0-preview.1/build/.NETFramework/v4.5.1</FrameworkPathOverride>
```
Comment (slang25) [inline src/Shouldly/Shouldly.csproj]: This property can go now, as it is disabled explicitly by the ReferenceAssemblies package

File: src/Shouldly/Shouldly.csproj [hunk 2]
```diff
@@ -34,7 +33,7 @@
   <ItemGroup Condition=" '$(TargetFramework)' == 'net451' ">
     <Reference Include="System" />
     <Reference Include="Microsoft.CSharp" />
-    <PackageReference Include="Microsoft.TargetingPack.NETFramework.v4.5.1" Version="1.0.1" ExcludeAssets="All" PrivateAssets="All" />
+    <PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies.net451" Version="1.0.0-preview.1" PrivateAssets="All" />
```
Comment (slang25) [inline src/Shouldly/Shouldly.csproj]: You can and probably should omit the `.net451` as it's intended to be referenced by the meta-package, the way NuGet works with meta-packages means that only `net451` will be resolved and downloaded.

File: src/Shouldly/Shouldly.csproj (hunk 1 above)
Comment (JosephWoodward) [reply src/Shouldly/Shouldly.csproj]: I did wonder this, but it blows up if I remove it.

Comment (slang25) [reply src/Shouldly/Shouldly.csproj]: Interesting, was that locally or on CI?

Comment (slang25) [reply src/Shouldly/Shouldly.csproj]: Just tried it and it works here. I did a write-up on how it works:
https://stu.dev/dotnet-framework-reference-assemblies-via-nuget/
`FrameworkPathOverride` should do nothing when this package is referenced.

Comment (JosephWoodward) [reply src/Shouldly/Shouldly.csproj]: > Interesting, was that locally or on CI?

Weird, tried it and it worked for some reason.

**Pull request merged**